crates/runtime/src/datafusion.rs

```diff
@@ -274,22 +277,28 @@ impl DataFusion {
             .await
             .context(UnableToExecuteQuerySnafu)?;
 
-        if let Some(cache_provider) = &self.cache_provider {
-            if let Some(cached_result) = cache_provider
-                .get(&plan)
-                .await
-                .context(FailedToAccessCacheSnafu)?
-            {
-                let record_batch_stream = Box::pin(
-                    MemoryStream::try_new(
-                        cached_result.records.to_vec(),
-                        cached_result.schema,
-                        None,
-                    )
-                    .context(UnableToCreateMemoryStreamSnafu)?,
-                );
+        let query_input_tables = get_logical_plan_input_tables(&plan);
+        let cache_enabled_for_query = !(query_input_tables.is_empty()
+            || query_input_tables.contains("information_schema.tables"));
```

Comment:
This string compare is likely to be expensive in the critical path. Isn't there any better way to detect this?

Reply:
Thank you, I'll update this.